src/language-js/print/assignment.js

```diff
@@ -243,6 +247,31 @@ function isAssignmentOrVariableDeclarator(node) {
   return isAssignment(node) || node.type === "VariableDeclarator";
 }
 
+function isCompletxTypeAliasParams(node) {
+  const typeParams = getTypeParametersFromTypeAlias(node);
+  if (isNonEmptyArray(typeParams)) {
+    if (
+      typeParams.length > 1 &&
+      Boolean(getLast(typeParams).default) &&
+      typeParams.every((param) => Boolean(param.constraint))
```

Comment:
Why check `.default` only for the last type parameters?
```suggestion
      Boolean(getLast(typeParams).default) &&
      typeParams.some((param) => param.constraint || param.default)
```

Reply:
I forgot about `param.bound`, thank you.

> Why check .default only for the last type parameter?

Because the default value of type parameters can only appear on last param.

Reply:
> Because the default value of type parameters can only appear on last param

It's not the case.

Reply:
Like this?

```ts
type Bar<T = string, S> = {}
```

Reply:
You're right, TS doesn't accept that. I thought `S` in that situation would default to `any`, but it doesn't work that way.

Reply:
Still, I think it's better to break the LHS even if at least one type parameter has a constraint or a default (`some`, not `every`).

Reply:
Okay, I have no strong opinon.